### PR TITLE
Adding autocnet

### PR DESCRIPTION
**Pull request by amystamile-usgs**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Kelvinrr** [comment]
I consent to be a maintainer for this recipe

**Comment by amystamile-usgs** [comment]
@conda-forge/help-python Not sure why the mac build is hanging.

File: recipes/autocnet/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      - matplotlib-base
+      - dill
+      - conda-forge::gdal
+      - geoalchemy2
+      - geopandas
+      - hoggorm
+      - imageio
+      - ipykernel
+      - kalasiris
+      - conda-forge::knoten
```
Comment (ocefpaf) [inline recipes/autocnet/meta.yaml:53]: In conda-forge, all packages are from conda-forge exclusively, no need this this syntax.

File: recipes/autocnet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      - networkx
+      - opencv
+      - plio
+      - plurmy
+      - pyyaml
+      - redis-py
+      - scikit-image
+      - scipy
+      - shapely
+      - sqlalchemy-utils
```
Comment (ocefpaf) [inline recipes/autocnet/meta.yaml:35]: All these should be in host, or just run. Build dependencies are for compilers and stuff that must be explicitly listed for cross-compiling.

File: recipes/autocnet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/DOI-USGS/Autocnet/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 233ec9b333305dbeb105a612a809b5cf8a4391c84d402ea92fe2615ba1cc290b
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/autocnet/meta.yaml:13]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

File: recipes/autocnet/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      - scikit-learn
+      - scipy
+      - shapely
+      - sqlalchemy
+      - sqlalchemy-utils
+      - redis-py <5
+      - conda-forge::usgscsm >=2.0.0
+      - vlfeat
+      - protobuf <=3.20
+
```
Comment (ocefpaf) [inline recipes/autocnet/meta.yaml:77]:
```suggestion
```

File: recipes/autocnet/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  sha256: 233ec9b333305dbeb105a612a809b5cf8a4391c84d402ea92fe2615ba1cc290b
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
+  noarch: python
+  number: 0
+
+requirements:
+    build:
+      - python >=3.10
```
Comment (ocefpaf) [inline recipes/autocnet/meta.yaml:19]:
```suggestion
```

File: recipes/autocnet/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    host:
+      - python >=3.10
+      - setuptools
```
Comment (ocefpaf) [inline recipes/autocnet/meta.yaml:22]:
```suggestion
      - pip
      - setuptools
```

**Comment by ocefpaf** [comment]
Not sure why Windows is hanging but b/c this is noarch we can merge it.

**Pull request merged**